src/config/mongoose_config_parser_toml.erl

```diff
@@ -1231,12 +1225,17 @@ mod_global_distrib_connections([<<"endpoint_refresh_interval_when_empty">>|_], V
     [{endpoint_refresh_interval_when_empty, V}];
 mod_global_distrib_connections([<<"disabled_gc_interval">>|_], V) ->
     [{disabled_gc_interval, V}];
-mod_global_distrib_connections([<<"tls">>|_] = _Path, false) ->
-    [{tls_opts, false}];
 mod_global_distrib_connections([<<"tls">>|_] = Path, V) ->
-    TLSOpts = parse_section(Path, V),
+    TLSOpts = parse_section(Path, V, fun format_global_distrib_tls/1),
     [{tls_opts, TLSOpts}].
 
+-spec format_global_distrib_tls([option()]) -> option().
+format_global_distrib_tls(Opts) ->
+    case proplists:lookup(enabled, Opts) of
+        {enabled, false} -> false;
+        _ -> Opts
```

Comment:
IMO it should be `false` by default, so that setting the opts with no `enabled = true` would not enable them.
```suggestion
        {enabled, true} -> maps:remove(enabled, Opts);
        _ -> false
```

Also: removed `enabled` to clean up the resulting opts.

Reply:
I figured somebody may omit `enabled` and provide other options, basically keeping the config as is right now. But it probably makes more sense as you suggest if we introduce the `enabled` option.